tmt/steps/execute/__init__.py

```diff
@@ -467,9 +470,24 @@ def check_abort_file(self, test: "tmt.Test", guest: Guest) -> bool:
             TMT_ABORT_SCRIPT.created_file).exists()
 
     @staticmethod
-    def test_duration(start: float, end: float) -> str:
+    def format_timestamp(timestamp: datetime.datetime) -> str:
+        """ Convert timestamp to a human readable format """
+
+        return timestamp.isoformat()
+
+    @staticmethod
+    def format_duration(duration: datetime.timedelta) -> str:
         """ Convert duration to a human readable format """
-        return time.strftime("%H:%M:%S", time.gmtime(end - start))
+
+        total_seconds = int(duration.total_seconds())
```

Comment:
IMO using `seconds` here would be better as it's always correct, unlike `total_seconds` which is not true after the first subtraction
also might make the last subtraction more readable

maybe also a helper function to do the division and multiplication/subtraction and return a tuple? either to return floor division result+remainder or already subtracted remaining part, which would make it pretty specific, not sure which I would prefer or if it even makes sense here for the 2 operations

Reply:
Dammit! I forgot we have `divmod()`...

Reply:
Fixed in e66aecf3:

* `divmod()`, because that's exactly what it's for,
* `counter` instead of `total_seconds` - `seconds` alone did not seem right, but I agree, `total_seconds` is also misleading,
* and there's a unit test for good measure.

Reply:
awesome, didn't know divmod existed and don't understand why when I was searching for something like that nothing showed up